invenio_rdm_pure/source/rdm/database.py

```diff
@@ -38,28 +39,33 @@ def select_query(self, fields: str, table: str, filters={}):
             return False
         return self.cursor.fetchall()
 
-    @staticmethod
-    def get_pure_user_id():
+    def get_pure_user_id(self):
         """Gets the userId of the Pure user.
 
         In case the user doesn't exist yet,
         it is created with credentials defined in config.py.
         """
-        datastore = current_app.extensions["security"].datastore
-        if datastore is not None:
-            invenio_pure_user_email = current_app.config.get("INVENIO_PURE_USER_EMAIL")
-            invenio_pure_user_password = current_app.config.get(
-                "INVENIO_PURE_USER_PASSWORD"
-            )
-            invenio_pure_user = datastore.get_user(invenio_pure_user_email)
-            if not invenio_pure_user:
-                invenio_pure_user = datastore.create_user(
-                    email=invenio_pure_user_email,
-                    password=hash_password(invenio_pure_user_password),
-                    active=True,
+        if not self.pure_user_id:
+            datastore = current_app.extensions["security"].datastore
+            if datastore is not None:
+                invenio_pure_user_email = current_app.config.get(
```

Comment:
it would be better to move those things also to the parameter list of the method. so to set default parameters in the method. so it is easier to test this method

Reply:
What do you have in mind specifically?

Reply:
i had in mind to move the current_app.config calls to the parameter default sets, but this is not possible, see you message above. 
i prefer to have pure functions, so to give those values as parameters to the function

Reply:
I have modified the function, I hope it is what you had in mind.